mllm/backends/cpu/CPUAllocator.cpp

```
@@ -18,12 +19,20 @@ void align_alloc(void** ptr, size_t required_bytes, size_t align) {
   }
+#if defined(MLLM_TRACY_ENABLE) && MLLM_TRACY_ENABLE == 1
+  TracyAlloc(p1, required_bytes + offset);
+#endif
   p2 = (void**)(((size_t)(p1) + offset) & ~(align - 1));  // NOLINT
   p2[-1] = p1;
   *ptr = p2;
 }
 
-void align_free(void* ptr) { free(((void**)ptr)[-1]); }
+void align_free(void* ptr) {
+  free(((void**)ptr)[-1]);
```

Clang (14.0.6)

[error] do not use pointer arithmetic

(cppcoreguidelines-pro-bounds-pointer-arithmetic,-warnings-as-errors)